### PR TITLE
[Admin UI Backend] Add API Endpoints to Approve/Deny Privacy Requests 

### LINKED ISSUE #244: [Admin UI] Approve / Deny Privacy Requests
### Is your feature request related to a specific problem?

Users should be able to approve or deny privacy requests after they are submitted.

### Describe the solution you'd like

- Add a config var (`REQUIRE_MANUAL_REQUEST_APPROVAL`) to indicate whether Fidesops should require requests submitted to be manually approved before dispatch.
- Update the `PrivacyRequest` creation endpoint to not automatically dispatch submitted requests into the `PrivacyRequestRunner`. Set `PrivacyRequest.status` to `pending`.
- Add extra statuses to the `PrivacyRequestStatus` class:
  - `approved`
  - `denied`
- Add fields `PrivacyRequest.approved_at` and `PrivacyRequest.approved_by`
- Add a new API view to approve or deny `PrivacyRequest`s.

```
PATCH /api/v1/privacy-requests/administrate/<approve/deny>

{
  "requestIds": [
    "id1",
    "id2",
    ...
  ]
}
```

- When a request is approved:
  - Set `PrivacyRequest.status` to `approved`,
  - Dispatch the request into the `PrivacyRequestRunner`.
- When a request is denied:
  - Set `PrivacyRequest.status` to `denied`.

### Questions

- How do we best record the audit trail here?
- ans: no additional work because 1 SR can only be approved/denied once

### PR DESCRIPTION
**Pull request by pattisdr**
❗  Contains database migration 

# Purpose

Users should be able to approve or deny privacy requests after they are submitted. Currently, if a privacy request is submitted, we dispatch it immediately but we need more options here.

# Changes

If `REQUIRE_MANUAL_REQUEST_APPROVAL ` is set (default = False), don't dispatch privacy request immediately.

- Add two separate endpoints to approve/deny the request.  
  - Approve dispatches the privacy request / marks privacy request as approved and sets `reviewed_by` as a FK to user
  - Privacy request must be in `pending` state to approve or deny
- Add two new statuses to PrivacyRequestStatus: `approved` and `denied`
- Add two new columns to PrivacyRequest - `reviewed_at` and `reviewed_by`

# Checklist

- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md))
- [x] Good unit test/integration test coverage
- [x] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #244

### CODE REVIEW

**Comment by pattisdr** [comment]
@nschoenbrot making a few changes after chatting with Sean, I'll let you know when this ready for review

File: src/fidesops/models/privacy_request.py [hunk 1]
```diff
@@ -91,6 +94,14 @@ class PrivacyRequest(Base):
...
         nullable=False,
     )
+    # When the request was approved/denied
+    reviewed_at = Column(DateTime(timezone=True), nullable=True)
+    # Who approved/denied the request
+    reviewed_by = Column(
+        String,
+        ForeignKey(FidesopsUser.id_field_path, ondelete="SET NULL"),
+        nullable=True,
+    )
```
Comment (pattisdr) [inline src/fidesops/models/privacy_request.py:104]: Currently, `PrivacyRequest.reviewed_by` is a FK to fidesopsuser.  Users can be deleted, which is an issue b/c of the FK.  I am currently just setting the `reviewed_by` field to None if the user is deleted, and leaving the privacy request in the db.

**Comment by pattisdr** [comment]
OK ready for review!

**Comment by conceptualshark** [comment]
👍 All looks good on the the docs side of things.

**Comment by pattisdr** [comment]
hey @conceptualshark, I didn't really add much in the way of docs here the first time, I think this was the day in standup we were told to stop doing that, but it sounds like having a draft is better?

Here's some guides for the endpoints being added:

https://github.com/ethyca/fidesops/pull/282/files#diff-80b27ef73b69abe08e0fe4243c3b4bf70fa1b5fb9bf9f6abc6c1c1a07db6832d

File: docs/fidesops/docs/postman/Fidesops.postman_collection.json [hunk 2]
```diff
@@ -728,7 +728,7 @@
 			"description": "These API calls step you through how to set up all the resources in Fidesops to execute erasure privacy requests in your system.\n\nIt assumes you've already run through the API requests in the \"Access\" section so database connections, storage, and annotated datasets are already configured."
 		},
 		{
-			"name": "Other Useful API calls",
+			"name": "Privacy Request Management",
```
[the remark this answers is not in the document]
Comment (galvana) [reply docs/fidesops/docs/postman/Fidesops.postman_collection.json:731]: Good call on moving these endpoints outside of a crowded folder. And thanks for the heads up on the diff!

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py [hunk 3]
```diff
@@ -455,3 +464,112 @@ def resume_privacy_request(
...
 
     return privacy_request
+
+
+def review_privacy_request(
+    db: Session, cache: FidesopsRedis, request_ids: List[str], process_request: Callable
+) -> BulkReviewResponse:
+    """Helper method shared between the approve and deny privacy request endpoints"""
+    succeeded: List[PrivacyRequest] = []
+    failed: [Dict[str, Any]] = []
```
Comment (galvana) [inline src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:474]: Should this be `List[Dict[str, Any]]` instead of `[Dict[str, Any]]`?

Comment (pattisdr) [reply src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:474]: whoops, thanks for catching!

File: docs/fidesops/docs/guides/privacy_requests.md [hunk 4]
```diff
@@ -57,6 +59,22 @@ we submit  a request to apply the `a-demo-policy` Policy to all target data in t
...
 
+## How can I approve or deny a Privacy Request?
+
+If `REQUEST_MANUAL_REQUEST_APPROVAL` is set to True (the default is False), Privacy Requests will require approval before executing.  To approve
+or deny Privacy Requests, send a list of Privacy Request ids to one of the following endpoints to approve/deny in bulk.
+
+`PATCH api/v1/privacy-request/administrate/approve`
+
+```json
+  {"request_ids": ["pri_2d181f15-486d-4bcf-a871-f50ed9f95673", "pri_2d181f15-486d-4bcf-a871-f50ed9f95673"]}
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply docs/fidesops/docs/guides/privacy_requests.md:70]: sounds good!

File: docs/fidesops/docs/guides/privacy_requests.md [hunk 5]
```diff
@@ -24,6 +25,7 @@ For more information on Policies, see [How-To: Configure Policies](policies.md#r
 
 You submit a Privacy Request by calling the  **Submit a Privacy Request** operation. Here, 
 we submit  a request to apply the `a-demo-policy` Policy to all target data in the [Identity Graph](../glossary.md) that can be generated from the email address `identity@example.com` and the phone number `+1 (123) 456 7891`.
+The Privacy Request will be executed immediately if `REQUEST_MANUAL_REQUEST_APPROVAL` is False (this is the default behavior).
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply docs/fidesops/docs/guides/privacy_requests.md]: that's nicer 👍

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py [hunk 6]
```diff
@@ -455,3 +464,112 @@ def resume_privacy_request(
...
+    """Helper method shared between the approve and deny privacy request endpoints"""
+    succeeded: List[PrivacyRequest] = []
+    failed: List[Dict[str, Any]] = []
+
+    for request_id in request_ids:
+        privacy_request = PrivacyRequest.get(db, id=request_id)
+        if not privacy_request:
+            failed.append(
+                {
+                    "message": f"No privacy request found with id '{request_id}",
```
Comment (galvana) [inline src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:481]: Missing second single quote to close out request_id.

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py [hunk 7]
```diff
@@ -455,3 +464,112 @@ def resume_privacy_request(
     ).submit(from_webhook=webhook)
 
     return privacy_request
+
+
+def review_privacy_request(
+    db: Session, cache: FidesopsRedis, request_ids: List[str], process_request: Callable
```
Comment (galvana) [inline src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:470]: I like the use of `process_request` here! Nice way to reuse the main error handling workflow.

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py [hunk 8]
```diff
@@ -182,10 +190,11 @@ def create_privacy_request(
                     for masking_secret in masking_secrets:
                         privacy_request.cache_masking_secret(masking_secret)
 
-            PrivacyRequestRunner(
-                cache=cache,
-                privacy_request=privacy_request,
-            ).submit()
+            if not config.execution.REQUIRE_MANUAL_REQUEST_APPROVAL:
```
Comment (galvana) [inline src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:193]: It looks like changes to this setting require a server restart to take effect. Do we foresee a need to have this change take effect without restarting the server?

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py (hunk 7 above)
Comment (pattisdr) [reply src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:470]: oh thank you!

File: src/fidesops/api/v1/endpoints/privacy_request_endpoints.py (hunk 8 above)
Comment (pattisdr) [reply src/fidesops/api/v1/endpoints/privacy_request_endpoints.py:193]: good thought, in my opinion, it's not a concern given our current number of users, but others may have different opinions

**Review by galvana — APPROVED** [review]
All conversations have been addressed or resolved, good to go!

**Pull request merged**